Support unenclosed inner text for details elements in to be visible

kschow opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

If you're new to contributing to open source projects, you might find this free
video course helpful: http://kcd.im/pull-request

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

**What**:

Supporting unenclosed inner text inside ```<details />``` elements for toBeVisible(). See https://github.com/testing-library/jest-dom/issues/395

**Why**:

Unenclosed inner text is valid for ```<details />``` elements and should therefore be supported.

**How**:

Updated the isAttributeVisible function. In the case where the inner text is unenclosed there is no "previousElement" so defaults to only check the open attribute of the details element itself. Otherwise, it follows the same logic as before.

**Checklist**:

<!-- Have you done all of these things?  -->

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [x] Documentation
- [x] Tests
- [ ] Updated Type Definitions "N/A"
- [x] Ready to be merged
      <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->

<!-- feel free to add additional comments -->

kschow commented on README.md:
```diff
@@ -49,7 +49,6 @@ clear to read and to maintain.
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
-
```
I think this, and the other formatting change might have been automatic, I didn't do them intentionally but I can try to revert those changes if necessary.

kschow commented on src/__tests__/to-be-visible.js:
```diff
@@ -247,6 +247,185 @@ describe('.toBeVisible', () => {
+          })
+        })
+
+        describe('with nested details (unenclosed outer, enclosed inner)', () => {
```
This set of tests might be a bit redundant, I added them because I wasn't sure how the unenclosed and enclosed details would interact. But thinking it over, there really shouldn't be a significant difference as opposed to the previous nested tests (which these are based off).

feedmypixel commented:
I've just come across this issue when testing details with `"@testing-library/jest-dom": "5.16.2"`. This PR looks a little stale? any help needed?

kschow commented:
I don't think I need any help with the actual code. There were a couple
lint issues that I ended up fixing and I don't think it re-ran the checks
to progress the status of the issue.

Feel free to do what you think is best here. It'd probably need to be
rebased or have the current master merged in because I'm sure things have
changed.

On Wed, Mar 30, 2022 at 12:00 PM Ben Chidgey ***@***.***>
wrote:

> I've just come across this issue when testing details with ***@***.***/jest-dom":
> "5.16.2". This PR looks a little stale? any help needed?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/testing-library/jest-dom/pull/396#issuecomment-1083390279>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAR2M2HXVULGVZQLNATWEK3VCSCBRANCNFSM5CU53EZA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

feedmypixel commented:
> Feel free to do what you think is best here. 

I have rebased main. All linting and testing is passing, can you add me as a collaborator on https://github.com/kschow/jest-dom/ and I will raise a PR to merge into your branch. Once you have merged that into yours that then should kick the status required checks here and we will be moving forward

feedmypixel commented:
Ok thats been raised on your repo and merged into your branch - i'll leave it up to you but thats your original PR with an updated main - its also kicked off the status checks here

kschow commented:
I think at this point we're waiting on a maintainer to take a look at this and approve it.
